src/main/java/com/github/thunkware/ThreadFactoryBuilderProvider.java

```diff
@@ -0,0 +1,93 @@
+package com.github.thunkware;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * Defines static methods to create virtual thread factories builders.
+ */
+public interface ThreadFactoryBuilderProvider {
+
+  /**
+   * Creates a new {@link VirtualThreadFactoryBuilder}
+   * 
+   * @return new instance of VirtualThreadFactoryBuilder
+   */
+  VirtualThreadFactoryBuilder ofVirtual();
+
+  /**
+   * Builder to create Virtual Threads Factories
+   */
+  interface VirtualThreadFactoryBuilder {
```

Comment:
Java 21 defines the virtual thread builder as [Thread.Builder.OfVirtual](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Thread.Builder.OfVirtual.html)

I think we should define this similarly as ThreadTool.Builder.OfVirtual. What do you think?